### PR TITLE
Add displayCheckoutBeforeConfirmation hook

### PR DESCRIPTION
**Pull request by Hlavtox**
| Questions         | Answers
| ------------------| -------------------------------------------------------
| Description?      | Add new https://github.com/PrestaShop/PrestaShop/pull/27437 hook during upgrade
| Type?             | new feature
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | -
| How to test?      | Install a new shop, make an upgrade to 8.0.0, check the hook is in DB
| Possible impacts? | No

### CODE REVIEW

**Review by kpodemski — DISMISSED** [review]

**Review by saulaski — DISMISSED** [review]

**Review by PululuK — APPROVED** [review]

**Comment by PululuK** [comment]
The main PR as been merger : https://github.com/PrestaShop/PrestaShop/pull/27437

**Review by okom3pom — APPROVED** [review]

**Comment by hibatallahAouadni** [comment]
Hello @Hlavtox 

Could you please provide steps to test the PR :pray: 
And please rebase your PR.

Thanks!

**Comment by Hlavtox** [comment]
Rebased

**Review by matks — DISMISSED** [review]

**Review by hibatallahAouadni — COMMENTED** [review]
Hello @Hlavtox 

I encountred an issue with your PR while upgrading my shop from 1.7.8.3 to 8.0.0:

`SQL 8.0.0 1064 in INSERT IGNORE INTO `ps_hook` (`id_hook`, `name`, `title`, `description`, `position`) VALUES (NULL, 'actionValidateOrderAfter', 'New Order', 'This hook is called after validating an order by core', '1'), (NULL, 'actionAdminOrdersTrackingNumberUpdate', 'After setting the tracking number for the order', 'This hook allows you to execute code after the unique tracking number for the order was added', '1'), (NULL, 'actionAdminSecurityControllerPostProcessBefore', 'On post-process in Admin Security Controller', 'This hook is called on Admin Security Controller post-process before processing any form', '1'), (NULL, 'actionAdminSecurityControllerPostProcessGeneralBefore', 'On post-process in Admin Security Controller', 'This hook is called on Admin Security Controller post-process before processing the General form', '1'), (NULL, 'dashboardZoneThree', 'Dashboard column three', 'This hook is displayed in the third column of the dashboard', '1'), (NULL, 'actionPresentPaymentOptions', 'Payment options Presenter', 'This hook is called before payment options are presented', '1'), (NULL, 'actionCustomerLogoutBefore', 'Before customer logout', 'This hook allows you to execute code before customer logout', '1'), (NULL, 'actionCustomerLogoutAfter', 'After customer logout', 'This hook allows you to execute code after customer logout', '1'), (NULL, 'displayCheckoutSummaryTop', 'Cart summary top', 'This hook allows you to display new elements in top of cart summary', '1') (NULL, 'displayCheckoutBeforeConfirmation', 'Show custom content before checkout confirmation', 'This hook allows you to display custom content at the end of checkout process', '1'): You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '(NULL, 'displayCheckoutBeforeConfirmation', 'Show custom content before checkout' at line 11`

But it seems the upgrade is done:

image
 
Then, when I tried to log in, I get this :trollface: 

image

Waiting for your feedback.
Thanks!

File: upgrade/sql/8.0.0.sql [hunk 1]
```diff
@@ -24,6 +24,7 @@ INSERT IGNORE INTO `PREFIX_hook` (`id_hook`, `name`, `title`, `description`, `po
   (NULL, 'actionCustomerLogoutBefore', 'Before customer logout', 'This hook allows you to execute code before customer logout', '1'),
   (NULL, 'actionCustomerLogoutAfter', 'After customer logout', 'This hook allows you to execute code after customer logout', '1'),
   (NULL, 'displayCheckoutSummaryTop', 'Cart summary top', 'This hook allows you to display new elements in top of cart summary', '1')
```
Comment (kpodemski) [inline upgrade/sql/8.0.0.sql:26]:
```suggestion
  (NULL, 'displayCheckoutSummaryTop', 'Cart summary top', 'This hook allows you to display new elements in top of cart summary', '1'),
```

whoops

**Review by kpodemski — CHANGES_REQUESTED** [review]
I think you missed the comma :)

**Review by kpodemski — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Comment by hibatallahAouadni** [comment]
Hello @Hlavtox 

Thanks for fixing the issue :pray: 
The upgrade is done with success, see the attached screenshot below:

image

image

LGTM, QA :heavy_check_mark: 

Thanks!

**Comment by Progi1984** [comment]
Thanks @Hlavtox & @hibatallahAouadni

**Comment by Hlavtox** [comment]
Yaaaaay

**Pull request merged**